system/DataCaster/DataCaster.php

```diff
@@ -113,12 +119,16 @@ public function setTypes(array $types): static
      * Add ? at the beginning of the type (i.e. ?string) to get `null`
      * instead of casting $value when $value is null.
      *
-     * @param mixed       $value  The value to convert
-     * @param string      $field  The field name
-     * @param 'get'|'set' $method Allowed to "get" and "set"
+     * @param mixed  $value  The value to convert
+     * @param string $field  The field name
+     * @param string $method Allowed to "get" and "set"
      */
     public function castAs(mixed $value, string $field, string $method = 'get'): mixed
     {
+        if ($method !== 'get' && $method !== 'set') {
```

Comment:
how about using `! in_array($method, ['set', 'get'], true))` ?

```suggestion
        if (! in_array($method, ['set', 'get'], true)) {
```